k4GaudiPandora/src/DDCaloHitCreator.cc

```
@@ -211,158 +203,146 @@ pandora::StatusCode DDCaloHitCreator::CreateECalCaloHits(const EVENT::LCEvent* c
             caloHitParameters.m_cellSize1 = splitCellSize;
           }
 
-          PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
-                                  PandoraApi::CaloHit::Create(m_pandora, caloHitParameters));
-          m_calorimeterHitVector.push_back(pCaloHit);
+          PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, PandoraApi::CaloHit::Create(
+            m_pandora, caloHitParameters));
+          m_calorimeterHitVector.push_back(caloID);
 
         } catch (pandora::StatusCodeException& statusCodeException) {
-          streamlog_out(ERROR) << "Failed to extract ecal calo hit from " << *iter << ": "
-                               << statusCodeException.ToString() << std::endl;
-        } catch (EVENT::Exception& exception) {
-          streamlog_out(WARNING) << "Failed to extract ecal calo hit from " << *iter << ": " << exception.what()
-                                 << std::endl;
+          log << MSG::ERROR << "Failed to extract ecal calo hit from " << pCaloHitCollection << ": "
+                               << statusCodeException.ToString() << endmsg;
         }
       }
-    } catch (EVENT::Exception& exception) {
-      streamlog_out(MESSAGE) << "Failed to extract ecal calo hit collection: " << *iter << ", " << exception.what()
-                             << std::endl;
+    } catch (...) {
+      log << MSG::ERROR << "Failed to extract ecal calo hit collection" << endmsg;
```

I would be interested in seeing which exceptions we get with this catch-all, resp. which one we should let propagate. In any case: Are there cases where catching an exception is a recoverable error? Or should we return a failure status code from here?

---

Again, as above, probably need to remove a good number of these try-catches.